motr/io_req_fop.c

```diff
@@ -101,6 +101,143 @@ M0_INTERNAL struct m0_file *m0_client_fop_to_file(struct m0_fop *fop)
 	return &ioo->ioo_flock;
 }
 
+/**
+ * Copies correct part of attribute buffer to client's attribute bufvec.
+ */
+
+/**
+ * Populates client application's attribute bufvec from the attribute buffer
+ * received from reply fop.
+ *
+ *     | CS0 | CS1 | CS2 | CS3 | CS4 | CS5 | CS6 |
+ *
+ *  1. rep_ivec* will be COB offset received from target
+ *     | rep_index[0] || rep_index[1] || rep_index[2] |
+ *     
+ *  2. ti_ivec will be COB offset while sending 
+ *     | ti_index[0] || ti_index[1] || ti_index[2] || ti_index[3] |     
+ *     *Note: rep_ivec will be subset of ti_ivec
+ * 
+ *  3. ti_goff_ivec will be GOB offset while sending 
+ *     Note: rep_ivec will be subset of ti_ivec
+ *
+ * Steps:
+ *   1. Bring reply ivec to start of unit
+ *   2. Then move cursor of ti_vec so that it matches rep_ivec start.
+ *      Move the ti_goff_ivec for by the same size
+ *      Note: This will make all vec aligned 
+ *   3. Then iterate over rep_ivec one unit at a time till we process all 
+ *      extents, get corresponding GOB offset and use GOB Offset and GOB 
+ *      Extents (ioo_ext) to locate the checksum add and copy one checksum
+ *      to the application checksum buffer.
+ * @param rep_ivec m0_indexvec representing the extents spanned by IO.
+ * @param ti       target_ioreq structure for this reply's taregt.
+ * @param ioo      Object's context for client's internal workings.
+ * @param buf      buffer contaiing attributes received from server.
+ */
+static void application_attribute_copy(struct m0_indexvec *rep_ivec,
+				       struct target_ioreq *ti,
+				       struct m0_op_io *ioo,
+				       struct m0_buf *buf)
+{
+	uint32_t                unit_size, off, cs_sz;
+	m0_bindex_t             rep_index;
+	m0_bindex_t             ti_cob_index;
+	m0_bindex_t             ti_goff_index;
+	struct m0_ivec_cursor   rep_cursor;
+	struct m0_ivec_cursor   ti_cob_cursor;
+	struct m0_ivec_cursor   ti_goff_cursor;
+	struct m0_indexvec     *ti_ivec = &ti->ti_ivec;
+	struct m0_indexvec     *ti_goff_ivec = &ti->ti_goff_ivec;
+
+	void *dst = ioo->ioo_attr.ov_buf[0];
+	void *src = buf->b_addr;
+
+	if(!buf->b_nob)
+	{
+		/* Return as no checksum is present */
+		return;
+	}
+	
+	unit_size = m0_obj_layout_id_to_unit_size(m0__obj_lid(ioo->ioo_obj));
+	cs_sz = ioo->ioo_attr.ov_vec.v_count[0];
+
+	m0_ivec_cursor_init(&rep_cursor, rep_ivec);
+	m0_ivec_cursor_init(&ti_cob_cursor, ti_ivec);
+	m0_ivec_cursor_init(&ti_goff_cursor, ti_goff_ivec);
+	
+	rep_index 	= m0_ivec_cursor_index(&rep_cursor);
+    ti_cob_index 	= m0_ivec_cursor_index(&ti_cob_cursor);
+	ti_goff_index 	= m0_ivec_cursor_index(&ti_goff_cursor); 
+
+	/* Move rep_cursor on unit boundary */
+	off = rep_index % unit_size; 	
+	if(off)
+	{
```

Comment:
Please change the style if-else block according to motr's coding style guidelines.